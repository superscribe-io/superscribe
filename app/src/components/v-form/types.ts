import { DeepPartial, Field } from '@directus/types';

export type FormField = DeepPartial<Field> & {
	field: string;
	name: string;
	hideLabel?: boolean;
	hideLoader?: boolean;
};
