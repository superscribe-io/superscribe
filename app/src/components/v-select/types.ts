export type Option = {
	value: string | number | null;
	icon?: string;
	text?: string;
	disabled?: boolean;
	children?: Option[];
	divider?: boolean;
	selectable?: boolean;
	hidden?: boolean;
};
