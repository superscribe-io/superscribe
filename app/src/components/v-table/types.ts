export type Alignment = 'left' | 'center' | 'right';

export type HeaderRaw = {
	text: string;
	value: string;
	description?: string | null;
	align?: Alignment;
	sortable?: boolean;
	width?: number | null;
	[key: string]: any;
};

export type Header = Required<HeaderRaw>;

export type Item = {
	[key: string]: any;
};

export type ItemSelectEvent = {
	value: boolean;
	item: Item;
};

export type Sort = {
	by: string | null;
	desc: boolean;
};
