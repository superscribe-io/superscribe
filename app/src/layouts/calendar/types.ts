export type LayoutOptions = {
	template?: string;
	startDateField?: string;
	endDateField?: string;
	firstDay?: number;
	viewInfo?: {
		type: string;
		startDateStr: string;
	};
};
