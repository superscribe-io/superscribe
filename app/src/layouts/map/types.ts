import { CameraOptions } from 'maplibre-gl';

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	limit: number;
	page: number;
};

export type LayoutOptions = {
	cameraOptions?: CameraOptions & { bbox: any };
	geometryField?: string;
	autoLocationFilter?: boolean;
	clusterData?: boolean;
	animateOptions?: any;
	displayTemplate?: string;
};
