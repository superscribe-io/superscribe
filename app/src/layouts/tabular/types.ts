export type LayoutOptions = {
	widths?: {
		[field: string]: number;
	};
	align?: {
		[field: string]: 'left' | 'center' | 'right';
	};
	limit?: number;
	spacing?: 'comfortable' | 'cozy' | 'compact';
};

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	page: number;
	limit: number;
};
