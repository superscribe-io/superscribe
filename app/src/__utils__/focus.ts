import { Directive } from 'vue';

export const Focus: Directive = {
	mounted(el, binding) {
		if (binding.value) {
			el.focus();
		} else {
			el.blur();
		}
	},
};
