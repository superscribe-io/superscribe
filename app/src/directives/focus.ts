import { Directive } from 'vue';

const Focus: Directive = {
	mounted(el, binding) {
		if (binding.value) {
			el.focus();
		} else {
			el.blur();
		}
	},
};

export default Focus;
