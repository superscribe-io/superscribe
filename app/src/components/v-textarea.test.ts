import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VTextarea from './v-textarea.vue';
import { Focus } from '@/__utils__/focus';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';

const global: GlobalMountOptions = {
	directives: {
		focus: Focus,
	},
};

test('Mount component', () => {
	expect(VTextarea).toBeTruthy();

	const wrapper = mount(VTextarea, {
		props: {
			modelValue: `This is the first paragraph.
This is the second one.`,
		},
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('stlye props', async () => {
	const types = ['expand-on-focus', 'full-width', 'disabled'];

	for (const type of types) {
		const wrapper = mount(VTextarea, {
			props: {
				[type]: true,
			},
			global,
		});

		expect(wrapper.classes()).toContain(type);
	}
});

test('trim prop', async () => {
	const wrapper = mount(VTextarea, {
		props: {
			modelValue: '  This is an info  ',
			trim: true,
		},
		global,
	});

	await wrapper.get('textarea').trigger('blur');

	expect(wrapper.emitted()['update:modelValue'][0]).toEqual(['This is an info']);
});

test('nullable prop', async () => {
	const wrapper = mount(VTextarea, {
		props: {
			modelValue: '  This is an info  ',
			nullable: true,
		},
		global,
	});

	await wrapper.get('textarea').setValue('');

	expect(wrapper.emitted()['update:modelValue'][0]).toEqual([null]);
});
