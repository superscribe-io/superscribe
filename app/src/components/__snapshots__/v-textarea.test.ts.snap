// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-6a3718b7=\\"\\" class=\\"v-textarea full-width has-content\\">
  <!--v-if--><textarea data-v-6a3718b7=\\"\\"></textarea>
  <!--v-if-->
</div>"
`;
